modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C

```diff
@@ -127,17 +118,21 @@ AbaqusUMATStress::computeProperties()
   _aqNOEL = _current_elem->id();
 
   // characteristic element length
-  // TODO: check what Abaqus actually passes in here (likely ~ element_volume^(1/element_dim))
-  _aqCELENT = std::numeric_limits<Real>::signaling_NaN();
+  _aqCELENT = std::pow(_current_elem->volume(), 1.0 / _current_elem->dim());
```

Comment:
Are the `sqrt()` and `cbrt()` functions more efficient? You'd only ever end up keeping the same value or taking the square or cube root. Maybe it's better to call those other functions.

Reply:
future optimization